packages/leancode_lint/lib/lints/add_sliver_prefix_for_widget_returning_sliver.dart

```diff
@@ -0,0 +1,97 @@
+import 'package:analyzer/dart/ast/ast.dart';
+import 'package:analyzer/error/listener.dart';
+import 'package:collection/collection.dart';
+import 'package:custom_lint_builder/custom_lint_builder.dart';
+import 'package:leancode_lint/helpers.dart';
+
+/// Displays warning for widgets which return slivers but do not have the
+/// `Sliver` or `_Sliver` prefix in their name.
+class AddSliverPrefixForWidgetReturningSliver extends DartLintRule {
+  AddSliverPrefixForWidgetReturningSliver() : super(code: _getLintCode());
+
+  static const ruleName = 'add_sliver_prefix_for_widget_returning_sliver';
+
+  @override
+  void run(
+    CustomLintResolver resolver,
+    ErrorReporter reporter,
+    CustomLintContext context,
+  ) {
+    context.registry.addClassDeclaration(
+      (node) {
+        final isWidgetClass = _isWidgetClass(node);
+        if (!isWidgetClass) {
+          return;
+        }
+
+        final startsWithSliver = _hasSliverPrefix(node.name.lexeme);
+        if (startsWithSliver) {
+          return;
+        }
+
+        final buildMethod = _getBuildMethod(node);
+        if (buildMethod == null) {
+          return;
+        }
+
+        // Get all return expressions from build method
+        final returnExpressions = switch (buildMethod.body) {
+          ExpressionFunctionBody(:final expression) => [expression],
+          BlockFunctionBody(:final block) =>
+            block.statements.expand(getAllInnerReturnStatements).toList(),
+          _ => <Expression>[],
+        };
+
+        final isSliver = _anyIsSliver(returnExpressions.whereNotNull());
```

Comment:
Use dart built in alternative
```suggestion
        final isSliver = _anyIsSliver(returnExpressions.nonNulls);
```